match.c

```diff
@@ -426,8 +428,9 @@ ZLIB_INTERNAL unsigned longest_match(deflate_state *const s, IPos cur_match) {
         scan += 2, match+=2;
         Assert(*scan == *match, "match[2]?");
         do {
-            unsigned long sv = *(unsigned long*)(void*)scan;
-            unsigned long mv = *(unsigned long*)(void*)match;
+            unsigned long sv, mv;
+            memcpy(&sv, scan, sizeof(sv)); // = *(unsigned long*)(void*)scan;
+            memcpy(&mv, match, sizeof(mv)); // = *(unsigned long*)(void*)match;
```

Comment:
This code is invalid in C... memcpy should be after "unsigned long xor" is declared.